internal/healthcheck/healthcheck.go

```diff
@@ -0,0 +1,140 @@
+package healthcheck
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+
+	"github.com/chia-network/go-chia-libs/pkg/rpc"
+	"github.com/chia-network/go-chia-libs/pkg/types"
+)
+
+// Healthcheck is the main container for the app
+type Healthcheck struct {
+	healthcheckPort uint16
+	client          *rpc.Client
+
+	// Last block height we received
+	lastHeight uint32
+
+	// Time we received the last block height
+	lastHeightTime time.Time
+}
+
+// NewHealthcheck returns a new instance of healthcheck
+func NewHealthcheck(port uint16, logLevel log.Level) (*Healthcheck, error) {
+	var err error
+
+	healthcheck := &Healthcheck{
+		healthcheckPort: port,
+	}
+
+	log.SetLevel(logLevel)
+
+	healthcheck.client, err = rpc.NewClient(rpc.ConnectionModeWebsocket, rpc.WithAutoConfig(), rpc.WithBaseURL(&url.URL{
+		Scheme: "wss",
+		Host:   viper.GetString("hostname"),
+	}))
+	if err != nil {
+		return nil, err
+	}
+
+	return healthcheck, nil
+}
+
+// OpenWebsocket sets up the RPC client and subscribes to relevant topics
+func (h *Healthcheck) OpenWebsocket() error {
+	err := h.client.Subscribe("metrics")
+	if err != nil {
+		return err
+	}
+
+	err = h.client.AddHandler(h.websocketReceive)
+	if err != nil {
+		return err
+	}
+
+	h.client.AddDisconnectHandler(h.disconnectHandler)
+	h.client.AddReconnectHandler(h.reconnectHandler)
+
+	return nil
+}
+
+// CloseWebsocket closes the websocket connection
+func (h *Healthcheck) CloseWebsocket() error {
+	//return m.client.DaemonService.CloseConnection()
+	return nil
```

Comment:
Commented line seems _maybe_ more important than it currently is? Wonder if this function is needed at all in its current state?

Reply:
The close connection command used to be connected, then was sort of pulled out when we refactored for websockets.. This is actually just a copy/paste command from chia-exporter, but its likely not actually going to be needed, so I can just delete it completely if that makes you like it more

Reply:
If you think it's possible this ever gets implemented in the future, you can ignore this comment 😄